src/kernel/hardware/keyboard.c

```diff
@@ -49,38 +64,104 @@ void init_keyboard(){
         kernel_settings.keyboard.translation_enabled = false;        
     }
     buf_position = 0;
+    
+}
+
+uint8_t update_modifiers(key_modifiers modifier, bool is_pressed) {
+    printf("modifier: %x\n", modifier);
+    if ( is_pressed == true ) {
+        current_modifiers |= modifier;
+    } else {
+        current_modifiers = current_modifiers  & ~(modifier);
+    }
+    
+    return current_modifiers;
 }
 
 void handle_keyboard_interrupt() {
     
-    int scancode = inportb(KEYBOARD_ENCODER_PORT);
+    uint8_t scancode = inportb(KEYBOARD_ENCODER_PORT);
   
     if(kernel_settings.keyboard.translation_enabled == true || kernel_settings.keyboard.scancode_set) {
```

Comment:
code style: you check if translation_enable == true, but then do an implicit check with scancode_set. I'd say make then both implicit, or explicit.